format/format.go

```diff
@@ -235,12 +235,52 @@ func (p *Printer) applySpecialIndentRules(node *parse.ListNode) {
 			p.applySpecialLet(node.Nodes)
 		case IndentLetfn:
 			p.applySpecialLetfn(node.Nodes)
+		case IndentFor:
+			p.applySpecialFor(node.Nodes)
 		case IndentDeftype:
 			p.applySpecialDeftype(node.Nodes)
 		}
 	}
 }
 
+func (p *Printer) applySpecialForLet(nodes []parse.Node) {
+	i := -1
```

Comment:
Rather than managing this index manually, you could use the index provided by the `for` loop:

```go
for i, node := range nodes {
  ...
}
```

Reply:
I can't do that because I'm skipping some nodes.